corehq/sql_accessors/sql_templates/soft_undelete_forms.sql

```diff
@@ -0,0 +1,19 @@
+DROP FUNCTION IF EXISTS soft_undelete_forms(TEXT, TEXT[], TEXT);
+
+CREATE FUNCTION soft_undelete_forms(
+    p_domain TEXT,
+    form_ids TEXT[],
+    p_problem TEXT,
```

Comment:
nit: I would rename this to `reason` since it make is clearer what the purpose is. Also wondering if this info shouldn't go into an `XFormOperationSQL`.